docs/fastdds/xtypes/discovery_matching.rst

```diff
@@ -0,0 +1,102 @@
+Discovery process leverages the optional |TypeInformation-api| value to ensure compatibility between communicating endpoints.
+Upon the discovery of a remote endpoint, *eProsima Fast DDS* examines the |TypeInformation-api| to determine
+the availability of essential data type information.
+
+The process of checking the types is as follows:
+
+1. **Evaluation of TypeInformation**: If the discovered enpoint has transmitted |TypeInformation-api|,
+     *eProsima Fast DDS* evaluates whether all required types are already known.
+     In cases where some types are missing, an automatic process to discover the necessary types is initiated.
+     |TypeObject-api| is used to communicate with other participants and to discover the topic types asynchronously.
+     Matching of the discovered enpoint is deferred until all necessary types are known.
+
+2. **Fallback Mechanism**: In scenarios where the discovered endpoint has not transmitted |TypeInformation-api|,
+     *eProsima Fast DDS* resorts to a fallback mechanism.
+     This mechanism relys on the type name to verify if the type is already known within the system.
+     If the type name is recognized, matching occurs.
+
+.. _xtypes_discovery_typeinformation:
+
+TypeInformation
+---------------
+
+|TypeInformation-api| is a fundamental component of the *XTypes 1.3* specification, facilitating the efficient exchange
+of essential data type information between DDS endpoints.
+Instead of transmitting the entire |TypeObject|, endpoints share a compact representation containing the |TypeIdV1-api|
+of the respective data type.
+The exchanged |TypeIdV1-api| serves as unique identifiers for the associated data types, allowing endpoints to verify
+compatibility and also to facilitate subsequent retrieval of the necessary |TypeObject-api| for the data types needed,
+This mechanism enhances interoperability without burdening non-interested endpoints with unnecessary data.
+
+This approach significantly reduces network overhead, especially when communicating with endpoints that may
+not require the complete type definition.
+
+The contents of TypeInformation include:
+
+* |TypeIdV1-api|: This is the core component of TypeInformation.
+  It represents the unique identifier of the data type associated with the endpoint and
+  enables endpoints to identify and verify the compatibility of data types during communication.
+
+* Dependency Information: Optionally, TypeInformation may include dependency information, which
+  indicates other data types on which the current data type depends.
+  This information helps endpoints understand the data type hierarchy and its dependencies,
+  which facilitates comprehensive compatibility checks.
+  The decision as to which types to include in the dependency information is left to the implementation.
+
+|TypeInformation-api| is described in the annexes of `DDS-XTypes V1.3`_ document,
+please refer to this document for details.
```

Comment:
Again, too much uninteresting information from the point of view of the user. I suggest rewriting changing the focus of the text.

Reply:
Done in 1f40ec6245de6255b8f3189ef10ca2fdf1c17fea